pkg/commands/git_commands/status.go

```diff
@@ -51,6 +51,14 @@ func (self *StatusCommands) WorkingTreeState() enums.RebaseMode {
 	return enums.REBASE_MODE_NONE
 }
 
+func (self *StatusCommands) Rebasing() bool {
```

Comment:
Actually just realised we can see if `self.RebaseMode() == enums.REBASE_MODE_REBASING` which spares us from making an unrelated check on the merge state. With Merging() we actually already have IsInMergeState returning a boolean

Reply:
So... this?
```
func (self *StatusCommands) Rebasing() bool {
	rebaseMode, err := self.RebaseMode()
	return err == nil && rebaseMode != enums.REBASE_MODE_NONE
}
```

And this?
```
func (self *StatusCommands) Merging() bool {
	mergeState, err := self.IsInMergeState()
	return err == nil && mergeState
}
```

Reply:
The important question is what to do with errors. The only way an error can happen is if there's an error returned by `os.Stat` and I think given how rare that will be, we can just log the error and return a boolean rather than return both an error and a boolean.

That means we can have our RebaseMode method return just the mode (no error) and Rebasing() can return just a bool. And we can have IsInMergeState just return a bool and rename it to Merging().

Thoughts?

Reply:
That's why I sorta kinda ignored the error, even though it isn't really an erroneous state.

I'll add a logpoint and return some `bool`.